ADD: delayed_job_pools_per_machine config

grzegorzblaszczyk opened the pull request:
ADD: delayed_job_pools_per_machine config
UPDATE: README

rab commented on README.md:
```diff
@@ -79,6 +79,28 @@ set :delayed_job_queues, ['mailer','tracking']
+#   }
+# }
+
+# set :delayed_job_pools, {
```
Did you intend to duplicate these settings in the docs?
It seems like one would use `:delayed_job_pools_per_machine` in preference to `:delayed_job_pools` rather than use both. Was that not your expectation?

grzegorzblaszczyk replied:
Yes, I wanted to duplicate the example for `set :delayed_job_pools_per_machine` from `set :delayed_job_pools`. 
I will deduplicate it in the next PR. 
Corrected version is here: 
https://github.com/goodylabs/capistrano3-delayed-job/commit/8f3b5cb768d15ddf0d58614504f1d9f25c828d0c

rab commented on lib/capistrano/tasks/delayed_job.rake:
```diff
@@ -15,6 +15,10 @@ namespace :delayed_job do
     unless fetch(:delayed_job_pools).nil?
       args << fetch(:delayed_job_pools, {}).map {|k,v| "--pool='#{k}:#{v}'" }.join(' ')
     end
+    unless fetch(:delayed_job_pools_per_machine).nil?
```
This would obviously have to change if `:delayed_job_pools_per_machine` being used would _skip_ any settings in `:delayed_job_pools`

grzegorzblaszczyk replied:
Corrected version is here: 
https://github.com/goodylabs/capistrano3-delayed-job/commit/8f3b5cb768d15ddf0d58614504f1d9f25c828d0c

rab commented:
While it sounds like the setting and its usage is clear to *you*, I think that you need to explain a bit more in the `README.md` _how_ to use the new `delayed_job_pools_per_machine` and, in particular, that it is an *alternative* to the simpler `delayed_job_pools` which will *be ignored* if the former is used.

grzegorzblaszczyk commented:
@rab I have updated the `README.md` on how to use `:delayed_job_pools_per_server`. 
I have changed the name of variable from `:delayed_job_pools_per_machine` to `:delayed_job_pools_per_server`, so that it matches the convention of `server` variable globally used in Capistrano.

I have also added a sentence:
> If you use :delayed_job_pools_per_server, :delayed_job_pools will be ignored.

so it's clear now that it ignores `:delayed_job_pools`.

I hope it is clear now. 

If you have any further suggestions about `README.md`, please do not hesitate to ask as I would like this feature to be straightforward for everyone.

rab commented:
I'll cut a new version soon, too.

grzegorzblaszczyk commented:
Thanks, @rab !
